### PR TITLE
fixes issues with node v4

### LINKED ISSUE #6: Brunch config options
FYI, this repo may be obsolete.  You can use the brunch plugin config to force plugins to run (http://brunch.io/docs/config#-plugins-).  This would allow using `ng-annotate-brunch` or `ng-annotate-uglify-js-brunch` with your desired use case.

### PR DESCRIPTION
**Pull request by soumak77**
resolves https://github.com/hgwood/inject-this-brunch/issues/6

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -2,7 +2,6 @@
 
 const anymatch = require("anymatch")
 const ngAnnotate = require("ng-annotate")
-const assignToCopy = (...sources) => Object.assign(...sources)
```
Comment (hgwood) [inline index.js:5]: Actually this was meant to say `(...sources) => Object.assign({}, ...sources)`. Could you update your calls to add an empty object as the first argument?

**Review by hgwood — CHANGES_REQUESTED** [review]

File: index.js (hunk 1 above)
Comment (soumak77) [reply index.js:5]: done

File: index.js [hunk 2]
```diff
@@ -18,22 +17,22 @@ module.exports = class ngAnnotateCompiler {
   compile(file) {
     if (this.isIgnored(file.path)) return Promise.resolve(file)
     const options = withInFile(this.config, file.path)
-    const {src, map = file.map, errors} = ngAnnotate(file.data, options)
```
Comment (hgwood) [inline index.js:21]: This means `map` will be assigned to the `map` property of the returned value, or if that's `undefined`, to `file.map`. Your version always uses `file.map`. Can you fix it?

File: index.js [hunk 3]
```diff
@@ -18,22 +17,22 @@ module.exports = class ngAnnotateCompiler {
...
-    if (errors) return Promise.reject(errors)
-    return Promise.resolve(assignToCopy(file, {data: src, map}))
+    const results = ngAnnotate(file.data, options)
```
Comment (hgwood) [inline index.js]: Nitpick: maybe the name could be a little more descriptive? What about `annotated`?

**Review by hgwood — CHANGES_REQUESTED** [review]
Could you change `package.json` also to say the minimum Node version is now 4.0.0?

File: index.js [hunk 4]
```diff
@@ -33,6 +33,6 @@ function defaults(options, defaults) {
 function withInFile(options, inFile) {
   if (!options.map) return options
   return Object.assign({}, options, {
-    map: Object.assign({}, options.map, {inFile})
+    map: Object.assign({}, options.map, {inFile: inFile})
```
Comment (hgwood) [inline index.js]: This is supported by Node 4 http://node.green/#object-literal-extensions

**Review by hgwood — CHANGES_REQUESTED** [review]
Nearly there :)

**Review by hgwood — APPROVED** [review]

**Comment by hgwood** [comment]
Looks good. Let me just run it to be sure.

**Comment by soumak77** [comment]
thanks for the quick replies Hugo

**Comment by hgwood** [comment]
Works nice. Thanks to you!

**Comment by hgwood** [comment]
Change published in 4.1.0. https://www.npmjs.com/package/inject-this-brunch

**Pull request merged**